vscode-dotnet-runtime-library/src/Utils/WebRequestWorkerSingleton.ts

```diff
@@ -337,8 +340,8 @@ export class WebRequestWorkerSingleton
 
         if (this.proxySettingConfiguredManually(ctx) || discoveredProxy)
         {
-            this.proxyAgent = new HttpsProxyAgent(ctx.proxyUrl ?? discoveredProxy);
-            return this.proxyAgent;
+            const proxyAgent = new HttpsProxyAgent(ctx.proxyUrl ?? discoveredProxy);
```

Comment:
I think this means proxyAgent is never cached, right? Can you explain why that's positive?

Reply:
Correct. That is a positive bc the previous case is a bug. This class uses the singleton pattern which allows it to share a web cache amongst all of the code, and such that there is only one global web worker in the entire code. However, some code does not want to run under the proxy. Each request may have its own proxy setting, although unlikely, (so maybe we don't really want to do this perf hit for a bug fix that seems hard to actually manifest) you could configure the API to work as such. So, it is best to recreate this each time.

Reply:
That's a great question, btw :+1: